fix(linter/plugins): fix TS types

overlookmotel opened the pull request:
Fix the types. Once I actually used them with `defineRule`, discovered they weren't right.

camc314 commented:
### Merge activity

* **Sep 26, 2:29 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
fix(linter/plugins): fix TS types (#14162)

Fix the types. Once I actually used them with `defineRule`, discovered they weren't right.
